### PR TITLE
feat: configmap/secret enhancements

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

This PR Contains the changes of ConfigMap and Secret merge

Fixes # [AB#4120](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/4120), [AB#4121](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/4121)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Manually tested all the scenarios by creating different different cm and secret

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ConfigMapSecret/InfoIconWithTippy.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { ReactComponent as Question } from '../../assets/icons/ic-help-outline.svg'
+import { ReactComponent as QuestionFilled } from '../../assets/icons/ic-help.svg'
+import { TippyCustomized, TippyTheme } from '@devtron-labs/devtron-fe-common-lib'
+
+export default function InfoIconWithTippy({
+    infoText,
+    documentationLink,
+}: {
```
Comment (sohel-rp) [inline src/components/ConfigMapSecret/InfoIconWithTippy.tsx:9]: move to types

File: src/components/ConfigMapSecret/InfoIconWithTippy.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    infoText: string
+    documentationLink: string
+}) {
+    return (
+        <TippyCustomized
+            theme={TippyTheme.white}
+            className="w-300 h-100 fcv-5"
+            placement="right"
+            Icon={QuestionFilled}
+            heading={'API tokens'}
```
Comment (sohel-rp) [inline src/components/ConfigMapSecret/InfoIconWithTippy.tsx:19]: change to be attr

File: src/components/ConfigMapSecret/ConfigMapSecretDataEditorContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+                                    handleChange={handleSecretDataChange}
+                                    handleDelete={handleSecretDataDelete}
+                                />
+                            ))}
+                    </React.Fragment>
+                )
+            }
+        }
+
+        return (
```
Comment (sohel-rp) [inline src/components/ConfigMapSecret/ConfigMapSecretDataEditorContainer.tsx:231]: Break return in small functions

File: src/components/ConfigMapSecret/ConfigMapSecretForm.tsx [hunk 4]
```diff
@@ -0,0 +1,771 @@
...
+                        type: ConfigMapActionTypes.multipleOptions,
+                        payload: {
+                            cmSecretState: CM_SECRET_STATE.OVERRIDDEN,
+                        },
+                    })
+                }
+            }
+        }
+
+        async function handleSubmit(e) {
```
Comment (sohel-rp) [inline src/components/ConfigMapSecret/ConfigMapSecretForm.tsx:151]: Split lengthy function into small functions.

Comment (sohel-rp) [reply src/components/ConfigMapSecret/ConfigMapSecretForm.tsx:151]: Also, format files

File: src/components/ConfigMapSecret/Secret/secret.utils.tsx [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+                    esoYaml: YAML.stringify(configMapSecretData.esoSecretData),
+                },
+            })
+        }
+    } catch (err) {
+        showError(err)
+    }
+}
+
+export function handleSecretDataYamlChange(
```
Comment (sohel-rp) [inline src/components/ConfigMapSecret/Secret/secret.utils.tsx:386]: same here

File: src/components/ConfigMapSecret/ConfigMapSecretForm.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/ConfigMapSecret/ConfigMapSecretForm.tsx]: done

File: src/components/ConfigMapSecret/ConfigMapSecretDataEditorContainer.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/components/ConfigMapSecret/ConfigMapSecretDataEditorContainer.tsx]: done

File: src/components/ConfigMapSecret/InfoIconWithTippy.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/ConfigMapSecret/InfoIconWithTippy.tsx]: done

File: src/components/ConfigMapSecret/InfoIconWithTippy.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/ConfigMapSecret/InfoIconWithTippy.tsx]: done

**Review by sohel-rp — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**